make/compiler_flags

```diff
@@ -62,6 +62,23 @@ CXX_MAJOR := $(shell $(CXX) -dumpversion 2>&1 | cut -d'.' -f1)
 CXX_MINOR := $(shell $(CXX) -dumpversion 2>&1 | cut -d'.' -f2)
 
 
+ifneq (,$(findstring clang,$(CXX_TYPE)))
+	CXXFLAGS_OPTIM ?= -fvectorize -ftree-vectorize -fslp-vectorize -ftree-slp-vectorize -fno-standalone-debug -fstrict-return -ftrigraphs -fvisibility=hidden -fvisibility-inlines-hidden
+	ifeq ($(shell expr $(CXX_MAJOR) \>= 5), 1)
```

Comment:
This logic is for clang but you then check CXX_MAJOR is >=5 ? Can you elaborate on this?

Reply:
So this is a clang oddity. `clang-6` and `clang-7` both give `4.2.1` for `-dumpversion`. using `-v` instead of `-dumpversion` can give some weird results across operating systems and requires some extra trickery and gotchas to figure out. So while we could support `flto` by default with clang-7, clang-6 needs the `llvm-ar`*. And since we can't tell them apart I think it's just easier to only support flto by default for systems with clang > 7.

\* I thought this was true for all versions of clang, but for clang > 6 you can just use the system `ar`. I just changed the logic here so that we don't set the `AR` for clang when compiling with flto.